src/libtexture/texturesys.cpp

```diff
@@ -1239,17 +1248,30 @@ TextureSystemImpl::texture_lookup_nomip (TextureFile &texturefile,
 
 
 
-// Scale the derivs as dictated by 'width' and 'blur', and also make sure
+// Scale the derivs as dictated by 'width', and also make sure
 // they are all some minimum value to make the subsequent math clean.
 inline void
 adjust_width (float &dsdx, float &dtdx, float &dsdy, float &dtdy,
-              float swidth, float twidth)
+              float swidth, float twidth /*, float sblur=0, float tblur=0*/)
 {
     // Trust user not to use nonsensical width<0
     dsdx *= swidth;
     dtdx *= twidth;
     dsdy *= swidth;
     dtdy *= twidth;
+#if 0
+    // You might think that blur should be added to the original derivs and
+    // then just carry on. And sometimes it looks fine, but for some
+    // situations the results are absurdly wrong, as revealed by the unit
+    // test visualizations. I'm leaving this code here but disabled as a
+    // reminder to myself not to try it again. It's wrong.
+    if (sblur+tblur != 0.0f /* avoid the work when blur is zero */) {
+        dsdx += std::copysign (sblur, dsdx);
```

Comment:
I think the right math involves thinking about the two filter kernels as gaussians and looking for the right convolution of the two.

I think there is a closed form for this.

Reply:
I'm sure you're right, but I don't have the bandwidth to chase it down at the moment. Considering that nobody (not even the person who originally reported the problem) has mentioned it in the last 5 years and we've never noticed, I'm very tempted to just do this fix (which at least is a small step over the status quo) for now, and only go down the long rathole of trying to perfectly solve it if somebody either really seems to care or wants to work out the math themselves.

Reply:
So what say you? Is this ok as it stands?

Reply:
Yep LGTM, maybe put in a TODO comment about the Gaussian convolution view of the problem for whenever we do look into it again?

Reply:
sure